designs/2019-relative-plugin-loading/README.md

```diff
@@ -0,0 +1,169 @@
+
+### Allow disambiguation of plugin names
+
+Instead of raising an error for duplicate plugin names, we could provide a way to disambiguate rule references, as proposed in https://github.com/eslint/rfcs/pull/5. However, this would introduce a lot of complexity, and it's not clear how often users actually want to load two configs that depend on the same plugin. Raising an error for this case should allow us to compatibly introduce a disambiguation mechanism in the future, if needed.
+
+### Use a different strategy for determining whether two plugins conflict with each other
+
+A few possibilities include:
+
+* Instead of raising an error when two plugins with the same name are loaded, ESLint could simply pick one of the plugins to use, or try to merge the plugins. However, this could create confusing issues, because ESLint would be applying rule configurations for one version of a plugin to another version of a plugin, which could result in invalid configurations.
+* Instead of *always* raising an error when two plugins with the same name are loaded, another possibility would be to only raise an error *if the two plugins are in different locations on disk*. (For example, if the two plugins were the same package and were deduplicated by a package manager, they would be allowed.) However, this would create robustness issues for end users, because conflicts could suddenly arise as a result of a patch release in a plugin. (Also see the overview [here](https://gist.github.com/not-an-aardvark/169bede8072c31a500e018ed7d6a8915).)
+* We could allow multiple loads to the same plugin name in two different config files, as long as the config files were considered to be part of the same "package" (under the assumption that two files in the same "package" would always resolve the the same plugin). However, this could be confusing since it's not clear that there is a good working definition of a "package" for this case, and the problem it solves can already be worked around by shareable config authors.
+
+### Relax the filename requirements for path-loaded plugins
+
+This RFC requires path-loaded plugins to have names ending in `eslint-plugin-`. Alternatively, we could just use the last segment of the path to determine a plugin name without requiring it to end with `eslint-plugin-`. However, this could cause some confusion, e.g. if creates a config with `plugins: ["./foo/bar/index.js"]`, then the plugin would end up being called `index`. Given that users can control filenames in their own projects, it seems better to require users to make the name explicit, since this also reduces the liklihood of plugin name conflicts.
```

Comment:
After thinking about this more, it would be nice if `plugins: ["foo"]` always had the same behavior as `plugins: [require.resolve("eslint-plugin-foo")]`, since this would make it easier to normalize configs with absolute paths. But this seems difficult to accomplish because ESLint might not be able to determine that the name of the plugin is `foo` when given `require.resolve("eslint-plugin-foo")`. Maybe we could support some other way to provide both an absolute path and a name.

Reply:
I have an idea about local plugins: https://github.com/eslint/rfcs/blob/eslintrc-improvements/designs/2019-eslintrc-improvements/minor-03-plugin-renaming.md

Reply:
That seems like a good idea for local plugins. However, how would it save people from the "plugin conflict" error?